generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -1343,6 +1343,11 @@ private void generateVariableLengthFieldSize(FieldSpec field,
                         buffer.printf("int _sizeBeforeStruct = _size.totalSize();%n", field.camelCaseName());
                         buffer.printf("this.%s.addSize(_size, _cache, _version);%n", field.camelCaseName());
                         buffer.printf("int _structSize = _size.totalSize() - _sizeBeforeStruct;%n", field.camelCaseName());
+                        VersionConditional.forVersions(field.nullableVersions(), possibleVersions).
+                            ifMember(__ -> {
+                                buffer.printf("_structSize += 1;%n");
+                            }).
+                            generate(buffer);
```

Comment:
Don't we need to also change the code at L1336? I think that this is where the mistake is because we add one byte in all cases for nullable structs. It should be moved within the tagged/not-tagged branches in order to be correct. You did it for the tagged case. I suppose that we need to do the same for the not-tagged case.

I also wonder whether we should do it as follow:

```
buffer.printf("int _sizeBeforeStruct = _size.totalSize();%n", field.camelCaseName());
VersionConditional.forVersions(field.nullableVersions(), possibleVersions).
    ifMember(__ -> {
        buffer.printf("_size.addBytes(1);%n");
    }).
    generate(buffer);
buffer.printf("this.%s.addSize(_size, _cache, _version);%n", field.camelCaseName());
buffer.printf("int _structSize = _size.totalSize() - _sizeBeforeStruct;%n", field.camelCaseName());
buffer.printf("_size.addBytes(ByteUtils.sizeOfUnsignedVarint(_structSize));%n");
```

What do you think?

Reply:
I think I understand what you mean. The `+ 1` is due to the block at L1336 so we should move it into the size calculation below and clone it for the non-tagged case.